File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    productExceptSelf(nums) {
        const n = nums.length;
        const res = new Array(n);
        const prefix = new Array(n);
        const suffix = new Array(n);

        prefix[0] = 1;
        suffix[n-1] = 1;

        //populate prefix
        for(let i = 1; i < n; i++){
            prefix[i] = nums[i-1] * prefix[i-1];
        }

        //populate suffix
        for(let i = n - 2; i >= 0; i--){
            suffix[i] = suffix[i+1] * nums[i+1];
        }

        //populate res
        for (let i = 0 ; i < n; i++){
            res[i] = prefix[i] * suffix[i];
        }

        return res
    }
}

File: Data Structures & Algorithms/sort-colors/submission-1.java
class Solution {
    public void sortColors(int[] nums) {
        for (int i = 0 ; i < nums.length ; i++){
            int minIdx = i ; 

            for (int j = i + 1; j < nums.length; j++){
                if (nums[j] < nums[minIdx]){
                    minIdx = j;
                }
            }

            int temp = nums[i];
            nums[i] = nums[minIdx];
            nums[minIdx] = temp;
        }
    }
}